### DISCUSSION TITLE
Is it possible to merge schemas automatically without specifying an array in codegen.yml?

### DISCUSSION THREAD

**Discussion by Drakota** [dotansimha/graphql-code-generator#4295] [General] [answered]
Hi! In a project I have something like this:
```ts
// auth.ts
export default gql`
input AuthInput {
 something...
}
`;
```

```ts
// index.ts
const schema = gql`
type Query {
something...
}
`;
export default [schema, auth];
```
I have to do `schema: ["src/schema/index.ts", "src/schema/auth.ts"]` in my `codegen.yml` file to generate the types that are in auth.ts, is there a way to just do `schema: src/schema/index.ts` and have it automatically generate all the imported types in my index.ts file?

**Comment by ardatan** [comment, marked as answer]
By default, Codegen parses the code file to extract GraphQL SDL first. If there is no SDL found, it imports it as a Node module . As I see, you want to skip the first behavior. So if you add `noPluck: true` to your `config`, it will import all your schema from  `index.ts` instead of extracting SDL.

**Reply by Drakota** [reply]
Thanks that worked! Is there any downsides of doing it this way?

**Reply by ardatan** [reply]
No, this method just runs that file as a code, the other one parses the code file as an AST.
